src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecProc.java

```diff
@@ -34,29 +37,32 @@ public class ContainerExecProc extends Proc implements Closeable, Runnable {
     private final ExecWatch watch;
     private final OutputStream stdin;
 
+    private final PrintStream printStream;
+
     @Deprecated
     public ContainerExecProc(ExecWatch watch, AtomicBoolean alive, CountDownLatch finished,
             Callable<Integer> exitCode) {
-        this(watch, alive, finished, (OutputStream) null);
+        this(watch, alive, finished, (OutputStream) null, (PrintStream) null);
     }
 
     @Deprecated
     public ContainerExecProc(ExecWatch watch, AtomicBoolean alive, CountDownLatch finished,
             ByteArrayOutputStream error) {
-        this(watch, alive, finished, (OutputStream) null);
+        this(watch, alive, finished, (OutputStream) null, (PrintStream) null);
     }
 
     @Deprecated
     public ContainerExecProc(ExecWatch watch, AtomicBoolean alive, CountDownLatch finished, OutputStream stdin,
             ByteArrayOutputStream error) {
-        this(watch, alive, finished, stdin);
+        this(watch, alive, finished, stdin, (PrintStream) null);
     }
 
-    public ContainerExecProc(ExecWatch watch, AtomicBoolean alive, CountDownLatch finished, OutputStream stdin) {
+    public ContainerExecProc(ExecWatch watch, AtomicBoolean alive, CountDownLatch finished, OutputStream stdin, PrintStream printStream) {
         this.watch = watch;
         this.stdin = stdin == null ? watch.getInput() : stdin;
         this.alive = alive;
         this.finished = finished;
+        this.printStream = printStream;
```

Comment:
To avoid any NPE or null check in usage of this field.

```suggestion
        this.printStream = printStream == null ? NullPrintStream.NULL_PRINT_STREAM : printStream;
```